develop/blockchains/maintenance/storage-migrations.md

```diff
@@ -1,4 +1,192 @@
 ---
 title: Storage Migrations
-description: TODO
----
\ No newline at end of file
+description: Ensure smooth runtime upgrades with storage migrations, update data formats, and prevent errors. Learn when and how to implement migrations efficiently.
+---
+
+# Storage Migrations
+
+## Introduction
+
+Storage migrations are a crucial part of the runtime upgrade process. They allow you to update the storage layout of your blockchain, adapting to changes in the runtime. Whenever you change the encoding or data types used to represent data in storage, you'll need to provide a storage migration to ensure the runtime can correctly interpret the existing stored values in the new runtime state.
+
+Storage migrations must be executed precisely during the runtime upgrade process to ensure data consistency and prevent [runtime panics](https://doc.rust-lang.org/std/macro.panic.html){target=\_blank}. The migration code needs to run as follows:
+
+- After the new runtime is deployed
+- Before any other code from the new runtime executes
+- Before any [`on_initialize`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.Hooks.html#method.on_initialize){target=\_blank} hooks run
+- Before any transactions are processed
+
+This timing is critical because the new runtime expects data to be in the updated format. Any attempt to decode the old data format without proper migration could result in runtime panics or undefined behavior.
+
+## Storage Migration Scenarios
+
+A storage migration is necessary whenever a runtime upgrade changes the storage layout or the encoding/interpretation of existing data. Even if the underlying data type appears to still "fit" the new storage representation, a migration may be required if the interpretation of the stored values has changed.
+
+Storage migrations ensure data consistency and prevent corruption during runtime upgrades. Below are common scenarios categorized by their impact on storage and migration requirements:
+
+=== "Migration Required"
+
+    - Reordering or mutating fields of an existing data type to change the encoded/decoded data representation
+    - Removal of a pallet or storage item warrants cleaning up storage via a migration to avoid state bloat
+
+=== "Migration Not Required"
+
+    - Adding a new storage item would not require any migration since no existing data needs transformation
+    - Adding or removing an extrinsic introduces no new interpretation of preexisting data, so no migration is required
```

Comment:
In this case, where we’re explaining the reasoning behind why a migration is required or not, using tabs might imply that users need to choose between them, but they're not choosing, they're just learning at this point. Instead, presenting the options outside of tabs improves visibility and scan-ability, allowing users to easily see why a migration is needed or not.

Reply:
I've replaced this with nested list items. WDYT?